sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.Xunit.Legacy.cs

```diff
@@ -98,116 +98,132 @@ public void Fact14()
+        public void Theory1(int arg1) // Noncompliant {{Add at least one assertion to this test case.}}
 //                  ^^^^^^^
         {
             var x = 42;
         }
 
         [Theory]
-        public void Theory2()
+        [InlineData(1)]
+        public void Theory2(int arg1)
         {
-            var x = 42;
-            Assert.Equal(x, 42);
+            Assert.Equal(42, arg1);
         }
 
         [Theory]
-        public void Theory3()
+        [InlineData(1)]
+        public void Theory3(int arg1)
         {
-            var x = 42;
-            Xunit.Assert.Equal(x, 42);
+            Xunit.Assert.Equal(42, arg1);
         }
 
         [Theory]
-        public void Theory5()
+        [InlineData(1)]
+        public void Theory5(int arg1)
         {
-            var x = 42;
-            x.Should().Be(42);
+            arg1.Should().Be(42);
         }
 
         [Theory]
-        public void Theory6()
+        [InlineData(1)]
+        public void Theory6(int arg1)
         {
             Action act = () => { throw new Exception(); };
             act.ShouldThrow<Exception>();
         }
 
         [Theory]
-        public void Theory7()
+        [InlineData(1)]
+        public void Theory7(int arg1)
         {
             Action act = () => { throw new Exception(); };
             act.ShouldNotThrow<Exception>();
         }
 
         [Theory]
-        public void Theory8()
+        [InlineData(1)]
+        public void Theory8(int arg1)
         {
-            AssertSomething();
+            AssertSomething(arg1);
         }
 
         [Theory]
-        public void Theory9()
+        [InlineData(1)]
+        public void Theory9(int arg1)
         {
-            ShouldSomething();
+            ShouldSomething(arg1);
         }
 
         [Theory]
-        public void Theory10()
+        [InlineData(1)]
+        public void Theory10(int arg1)
         {
-            ExpectSomething();
+            ExpectSomething(arg1);
         }
 
         [Theory]
-        public void Theory11()
+        [InlineData(1)]
+        public void Theory11(int arg1)
         {
-            MustSomething();
+            MustSomething(arg1);
         }
 
         [Theory]
-        public void Theory12()
+        [InlineData(1)]
+        public void Theory12(int arg1)
         {
-            VerifySomething();
+            VerifySomething(arg1);
         }
 
         [Theory]
-        public void Theory13()
+        [InlineData(1)]
+        public void Theory13(int arg1)
         {
-            ValidateSomething();
+            ValidateSomething(arg1);
         }
 
         [Theory]
-        public void Theory14()
+        [InlineData(1)]
+        public void Theory14(int arg1)
         {
             dynamic d = 10;
             Assert.Equal(d, 10.0);
         }
 
-        public void AssertSomething()
+        [Theory(Skip = "reason")]
+        [InlineData(1)]
+        public void Theory15(int arg1) // Don't raise on skipped test methods
```

Comment:
Regression test for one of the bugs - _Skip_ was only being handled for _Fact_ tests.